### PR TITLE
2435 refactor Missiles part 1

### PR DESCRIPTION
**Pull request by bcko**
- related to #2435 
- part 1 of refactor Missiles
- remove global variable `Missiles` in function `UpdateMissileVelocity`

### CODE REVIEW

File: Source/missiles.cpp [hunk 1]
```diff
@@ -4354,95 +4379,97 @@ void MI_Element(int i)
...
 		PutMissile(i);
 	}
 }
 
 void MI_ResurrectBeam(int i)
 {
+	auto &missile = Missiles[i];
 	Missiles[i]._mirange--;
 	if (Missiles[i]._mirange == 0)
 		Missiles[i]._miDelFlag = true;
```
Comment (AJenbo) [inline Source/missiles.cpp:4475]: Looks like you ran out of steam here :)
```suggestion
	missile._mirange--;
	if (missile._mirange == 0)
		missile._miDelFlag = true;
```

**Review by AJenbo — APPROVED** [review]
Nice work, this is a really good clean up to have as it make it much easier to manage this code.

File: Source/missiles.cpp (hunk 1 above)
Comment (bcko) [reply Source/missiles.cpp]: 😄 updated. Thank you!

**Comment by AJenbo** [comment]
@bcko i'm going to see if i can simplify `AddMissile()` based on what you have done here it should be manageable now, hope it isn't conflicting with anything you are doing.

**Review by AJenbo — APPROVED** [review]

**Comment by bcko** [comment]
> @bcko i'm going to see if i can simplify `AddMissile()` based on what you have done here it should be manageable now, hope it isn't conflicting with anything you are doing.

sounds good! 👍

**Comment by AJenbo** [comment]
> sounds good! +1

Actually scratch that, it's premature of me to do this and you are doing a good job of this so please just continue, here is my wish for your next part in this series :)

Part 2:
`UpdateMissilePos()`
`SetMissAnim()`
`SetMissDir()`
`SetMissAnim()`
`MissilesFoundTarget()`
`UpdateVileMissPos()`
`AddLightningWall()` (you missed one usage here)

Part 3 (long list, but they should be easy, of cause feel free to split it up even further):
`AddRune()`
`AddImmolationRune()`
`AddStoneRune()`
`AddReflection()`
`AddBerserk()`
`AddJester()`
`AddStealPotions()`
`AddManaTrap()`
`AddSpecArrow()`
`AddMana()`
`AddRing()`
`AddMana()`
`AddArrow()`
`AddRndTeleport()`
`AddFirebolt()`
`AddMisexp()`
`AddWeapexp()`
`AddFlash()`
`AddFlash2()`
`AddGuardian()`
`AddChain()`
`AddAcidpud()`
`AddStone()`
`AddBoom()`
`AddHeal()`
`AddFirewallC()`
`AddInfra()`
`AddWave()`
`AddNova()`
`AddBlodboil()`
`AddApoca()`
`AddFlame()`
`AddResurrectBeam()`

Part 4:
`CheckMissileCol()`

**Comment by bcko** [comment]
@AJenbo thank you for writing a detailed guideline! I will work on part 2 and open a pull request.

**Pull request merged**